### PR TITLE
Development

### PR DESCRIPTION
**Pull request by DankeMary**

### CODE REVIEW

File: database/src/main/java/dao/AbstractDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    private final Class<T> clazz;
+
+    public AbstractDaoImpl(Class<T> clazz) {
+        this.clazz = clazz;
+        getEntityManager();
+    }
+
+    public void create(T t) {
+        getEntityManager().getTransaction().begin();
```
Comment (torgovtsev) [inline database/src/main/java/dao/AbstractDaoImpl.java]: please remove part with transaction.
doing it in all methods

File: database/src/main/java/dao/AbstractDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        getEntityManager().close();
+    }
+
+    public EntityManager getEntityManager() {
+        if (em == null)
+            em = EMFactory.getEntityManagerFactory().createEntityManager();
+        return em;
+    }
+
+    public EntityManager setEntityManager() {
```
Comment (torgovtsev) [inline database/src/main/java/dao/AbstractDaoImpl.java]: if method called with adv set please transfer to the method param
setEntityManager(EntityManaget) { this.entityManager = entityManager;}

File: database/src/main/java/dao/Dao.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package dao;
+
+import javax.persistence.EntityManager;
+
+public interface Dao {
+    EntityManager getEntityManager();
+    EntityManager setEntityManager();
```
Comment (torgovtsev) [inline database/src/main/java/dao/Dao.java]: if method called with adv set please transfer to the method param
setEntityManager(EntityManaget) { this.entityManager = entityManager;}

File: database/src/main/java/dao/EMFactory.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    static {
+        try {
+            Map<String, String> properties = new HashMap<String, String>();
+            properties.put("javax.persistence.jdbc.driver", "com.mysql.jdbc.Driver");
+            properties.put("javax.persistence.jdbc.url", "jdbc:mysql://localhost:3306/sbb_db");
+            properties.put("javax.persistence.jdbc.user", "root");
+            properties.put("javax.persistence.jdbc.password", "root");
+
+            emf = Persistence.createEntityManagerFactory("sbb", properties);
+        } catch (Throwable e) {
```
Comment (torgovtsev) [inline database/src/main/java/dao/EMFactory.java]: catch Exception, not Throwable

File: datatransferobjects/src/main/java/PassengerDTO.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import java.sql.Date;
+import java.util.Collection;
+
+public class PassengerDTO {
```
Comment (torgovtsev) [inline datatransferobjects/src/main/java/PassengerDTO.java]: Is it DTO object or it's SO object?
If it DTO please implement this object with immutable properties

File: datatransferobjects/src/main/java/PassengerDTO.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    private int id;
+    private String firstName;
+    private String lastName;
+    private Date birthday;
+    private Collection<TicketDTO> tickets;
+
+    public PassengerDTO(int id, String firstName, String lastName, Date birthday, Collection<TicketDTO> tickets) {
```
Comment (torgovtsev) [inline datatransferobjects/src/main/java/PassengerDTO.java]: Is this all parameters are mandatory?

File: serviceobjects/src/main/java/PassengerSO.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public Collection<TicketSO> getTickets() {
+        return tickets;
+    }
+    public void setTickets(Collection<TicketSO> tickets) {
+        this.tickets = tickets;
+    }
+
+    //TODO
```
Comment (torgovtsev) [inline serviceobjects/src/main/java/PassengerSO.java]: please provide author how made this TODO and short description what are you mean

File: serviceobjects/src/main/java/TripDataSO.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Collection;
+
+public class TripDataSO {
+    private int id;
+    private RouteSO route;
+    private int seatsLeft;
+    private byte isCancelled;
+    private byte isLate;
+    private Date date;
+    private Collection<TicketSO> ticketsFrom;
```
Comment (torgovtsev) [inline serviceobjects/src/main/java/TripDataSO.java]: please init this collection or modify get-method.
Collection property shouldn't be null

**Review by torgovtsev — COMMENTED** [review]
Please resolve all remarks.
Fill free to ask me you question.
Thanks

File: database/src/main/java/net/tsystems/EMFactory.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    static {
+        try {
+            Map<String, String> properties = new HashMap<String, String>();
+            properties.put("javax.persistence.jdbc.driver", "com.mysql.jdbc.Driver");
+            properties.put("javax.persistence.jdbc.url", "jdbc:mysql://localhost:3306/sbb_db");
+            properties.put("javax.persistence.jdbc.user", "root");
+            properties.put("javax.persistence.jdbc.password", "root");
+
+            emf = Persistence.createEntityManagerFactory("sbb", properties);
+        } catch (Throwable e) {
```
Comment (torgovtsev) [inline database/src/main/java/net/tsystems/EMFactory.java]: Don't catch Throwable, please catch Excpetion

File: database/src/main/java/net/tsystems/entitydao/PassengerDAO.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
+package net.tsystems.entitydao;
+
+import net.tsystems.entities.PassengerDO;
+
+import java.util.List;
+
+public interface PassengerDAO {
```
Comment (torgovtsev) [inline database/src/main/java/net/tsystems/entitydao/PassengerDAO.java]: You don't need all of those methods. The reason is that all of those methods are implementing into abstractDao

Comment (DankeMary) [reply database/src/main/java/net/tsystems/entitydao/PassengerDAO.java]: Then is it supposed to have only "specific" methods, like getAll, getByName, etc.? Basically everything that's not simple CRUD.

File: serviceobjects/src/main/java/PassengerSO.java (hunk 7 above)
Comment (DankeMary) [reply serviceobjects/src/main/java/PassengerSO.java]: TODO: add equals & hashCode
Done.

**Pull request merged**